repo.go

```diff
@@ -695,6 +696,41 @@ func (r *Repo) AddTargets(paths []string, custom json.RawMessage) error {
 	return r.AddTargetsWithExpires(paths, custom, data.DefaultExpires("targets"))
 }
 
+func (r *Repo) AddDigestTargets(digest string, length int64, custom json.RawMessage, path string) error {
+	expires := data.DefaultExpires("targets")
+
+	t, err := r.targets()
+	if err != nil {
+		return err
+	}
+	splitDigest := strings.Split(digest, ":")
+	if len(splitDigest) != 2 {
+		return fmt.Errorf("incorrect format for digest: %s", digest)
+	}
+	hash := splitDigest[1]
+	hashAlg := splitDigest[0]
+
+	meta := data.FileMeta{Length: length, Hashes: make(data.Hashes, 1)}
+	meta.Hashes[hashAlg], err = hex.DecodeString(hash)
+	if err != nil {
+		return err
+	}
+
+	if path == "" {
+		path = digest
+	}
```

Comment:
Should this default repo design be baked into the the go-tuf library? For callers of the library, it's just as easy to pass `digest` instead of an empty string `""`, while being more explicit/readable, and then the library doesn't need to maintain this API promise.